### PR TITLE
Fix displaying new product pages without redeploying

### PR DESCRIPTION
**Pull request by PaitoAnderson**
Test Procedure:
1. Run `yarn build`, `yarn start`.
2. Add a new product in chec portal.
3. Reload and navigate to new product permalink page.

### CODE REVIEW

File: pages/product/[permalink].js [hunk 1]
```diff
@@ -144,22 +130,21 @@ export async function getStaticPaths() {
...
-      product,
-    },
-  };
+  try {
+    const product = await commerce.products.retrieve(permalink, { type: 'permalink '});
+
+    // Pass product data to the page via props
+    return product ? { props: { product } } : { notFound: true };
+  } catch (error) {
+    console.error(error);
```
Comment (robbieaverill) [inline pages/product/[permalink].js:147]: Could you remove this line?

**Review by robbieaverill — COMMENTED** [review]
Looks good overall, thanks for the pull request 🙂

**Comment by PaitoAnderson** [comment]
Didn't realize we couldn't use `fallback: true` with `next export`. I'll have to rethink this one.

File: pages/product/[permalink].js (hunk 1 above)
Comment (PaitoAnderson) [reply pages/product/[permalink].js]: Yup, done

**Comment by robbieaverill** [comment]
Thanks @PaitoAnderson, we'll review this shortly

**Review by jaepass — APPROVED** [review]
Thanks Paito, tested and works great! Just a minor suggestion that you can also import `useState` from react and update the state definitions in the component.

**Comment by PaitoAnderson** [comment]
Thank you for reviewing @jaepass! Makes sense, updated. 

I also removed a `modules` prop I should never have added a few PRs back in the swiper.

**Pull request merged**